File: Leetcode 3501-3600/3507/3507.java
// Leetcode 3507: Minimum Pair Removal to Sort Array I
// https://leetcode.com/problems/minimum-pair-removal-to-sort-array-i/
// Solved on 22nd of January, 2026
class Solution {
    /**
     * Calculates the minimum number of operations to sort the array by repeatedly
     * replacing the pair of adjacent elements with the minimum sum with their sum.
     * @param nums An array of integers.
     * @return The minimum number of operations to sort the array.
     */
    public int minimumPairRemovals(int[] nums) {
        int n = nums.length;
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = nums[i];
        }

        int operations = 0;

        while (true) {
            boolean sorted = true;
            for (int i = 0; i < n - 1; i++) {
                if (arr[i] > arr[i + 1]) {
                    sorted = false;
                    break;
                }
            }

            if (sorted) {
                return operations;
            }

            int minSum = Integer.MAX_VALUE;
            int minIndex = -1;

            for (int i = 0; i < n - 1; i++) {
                int sum = arr[i] + arr[i + 1];
                if (sum < minSum) {
                    minSum = sum;
                    minIndex = i;
                }
            }

            arr[minIndex] = minSum;
            for (int i = minIndex + 1; i < n - 1; i++) {
                arr[i] = arr[i + 1];
            }
            n--;
            operations++;
        }
    }
}

File: Leetcode 3501-3600/3507/3507.py
# Leetcode 3507: Minimum Pair Removal to Sort Array I
# https://leetcode.com/problems/minimum-pair-removal-to-sort-array-i/
# Solved on 22nd of January, 2026
class Solution:
    def minimumPairRemoval(self, nums: list[int]) -> int:
        """
        Calculates the minimum number of pair removal operations to sort the array.

        Args:
            nums (list[int]): The input list of integers.

        Returns:
            int: The number of operations performed to make the array sorted.
        """
        operations = 0
        while True:
            isSorted = True
            for i in range(len(nums) - 1):
                if nums[i] > nums[i + 1]:
                    isSorted = False
                    break

            if isSorted:
                return operations

            minSum = float('inf')
            mergeIndex = -1

            for i in range(len(nums) - 1):
                currentSum = nums[i] + nums[i + 1]
                if currentSum < minSum:
                    minSum = currentSum
                    mergeIndex = i

            nums[mergeIndex] = minSum
            nums.pop(mergeIndex + 1)
            operations += 1